test/metabase/upload_test.clj

```diff
@@ -52,23 +112,36 @@
            ["2"          2              int-type]
            ["-86"        -86            int-type]
            ["9,986,000"  9986000        int-type]
-           [",,,"        nil            int-type]   ;; TODO: this should be a vchar in the future
-           ["9.986.000"  "9.986.000"    vchar-type] ;; TODO: this should be an integer in the future
+           ["9,986,000"  9986000        int-type "."]
+           ["9,986,000"  9986000        int-type ".,"]
+           ["9.986.000"  9986000        int-type ",."]
+           ["9’986’000"  9986000        int-type ".’"]
+           ["9.986.000"  "9.986.000"    vchar-type ".,"]
            ["3.14"       3.14           float-type]
-           [".14"        0.14           float-type]
-           ["0.14"       0.14           float-type]
-           ["-9986.567"  -9986.567      float-type]
-           ["$2.0"       2.0            float-type]
-           ["$ 3.50"     3.50           float-type]
-           ["-4300.23€"  -4300.23       float-type]
+           ["3.14"       3.14           float-type "."]
+           ["3.14"       3.14           float-type ".,"]
+           ["3,14"       3.14           float-type ",."]
+           ["3,14"       3.14           float-type ", "]
+           ["3.14"       3.14           float-type ".’"]
+           [".14"        ".14"          vchar-type ".,"] ;; TODO: this should be a float type
```

Comment:
The regex could do with visiting again to fix this case. I deliberately made this not parse as a float so that "." wouldn't be parsed as a float. But you could probably do something smarter than `#"\d[\d,]*\.\d+"`